ParseLoginUI/src/main/java/com/parse/ParseQueryRecyclerViewAdapter.java

```diff
@@ -0,0 +1,611 @@
+package com.parse;
+
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.support.v7.widget.RecyclerView;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentHashMap;
+
+import bolts.Capture;
+
+/**
+ * Created by lukask on 9/1/15.
+ */
+public class ParseQueryRecyclerViewAdapter<T extends ParseObject> extends RecyclerView.Adapter<ParseQueryRecyclerViewAdapter.ViewHolder> {
+
+    /**
+     * Implement to construct your own custom {@link ParseQuery} for fetching objects.
+     */
+    public interface QueryFactory<T extends ParseObject> {
+        ParseQuery<T> create();
+    }
+
+    /**
+     * Implement with logic that is called before and after objects are fetched from Parse by the
+     * adapter.
+     */
+    public interface OnQueryLoadListener<T extends ParseObject> {
+        void onLoading();
+
+        void onLoaded(List<T> objects, Exception e);
+    }
+
+    /**
+     * OnClickListener.
+     */
+    public interface OnClickListener<T extends ParseObject> {
+        void onClick(T item, int position);
+    }
+
+    /**
+     * ViewHolder.
+     */
+    public static class ViewHolder extends RecyclerView.ViewHolder {
```

Comment:
Are we going to allow subclassing of `ViewHolder`? If so, we may need `<T extends ViewHolder>` for the parent class.